packages/flutter/lib/src/painting/fractional_offset.dart

```diff
@@ -118,16 +124,16 @@ abstract class FractionalOffsetGeometry {
       return FractionalOffsetDirectional.lerp(a, b, t);
     if (a == null) {
       return new _SchrodingersFractionalOffset(
-        ui.lerpDouble(0.5, b._dxForRTL, t),
-        ui.lerpDouble(0.5, b._dxForLTR, t),
-        ui.lerpDouble(0.5, b._dy, t),
+        ui.lerpDouble(0.0, b._dxForRTL, t),
```

Comment:
why not 1.0?

Reply:
It's using `FractionalOffset.topLeft` for `null`, which is zeros across the board.  Did you want me to use `FractionalOffset.topStart` instead?  That's a little stranger mathematically because it's not the same origin as `operator*`.

Reply:
By the way, there's another way to handle null: we can treat it as being equal to whatever you're lerping against.  That's more different from how the other lerps work, but might be cleaner given that we already consider the type of whatever you're lerping against.

Reply:
Hmm, I thought this would work but the more I think about this the more concerned I am that `* 0.0` doesn't bring this to topStart... I need to think about this more.

Reply:
Treating `null` as center has the advantage that it's a point that's representable both as a `FractionalOffset` and a `FractionalOffsetDirectional`.  That makes the `lerp` less dependent on the types of its arguments.

Reply:
I'd be fine with `FractionalOffsetGeometry.lerp(null, FractionalOffset)` lerping from topStart and `FractionalOffset.lerp(null, FractionalOffset)` lerping from topLeft. My bigger concern is over `operator *`, especially cases like `FractionalOffset.topEnd * 0.0 + FractionalOffset.topRight * 0.0` (which should IMHO be the same as `FractionalOffset.topStart + FractionalOffset.topLeft`) vs  `(FractionalOffset.topEnd + FractionalOffset.topRight) * 0.0` (which should be the same).

Reply:
Let's move this discussion to the bug.